src/test/kotlin/com/pubnub/api/legacy/endpoints/files/PublishFileMessageTest.kt

```diff
@@ -0,0 +1,92 @@
+package com.pubnub.api.legacy.endpoints.files
+
+import com.github.tomakehurst.wiremock.client.WireMock.aResponse
+import com.github.tomakehurst.wiremock.client.WireMock.get
+import com.github.tomakehurst.wiremock.client.WireMock.getRequestedFor
+import com.github.tomakehurst.wiremock.client.WireMock.stubFor
+import com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo
+import com.github.tomakehurst.wiremock.client.WireMock.verify
+import com.github.tomakehurst.wiremock.matching.AbsentPattern
+import com.github.tomakehurst.wiremock.matching.EqualToPattern
+import com.pubnub.api.SpaceId
+import com.pubnub.api.endpoints.files.PublishFileMessage
+import com.pubnub.api.legacy.BaseTest
+import com.pubnub.api.models.consumer.MessageType
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+
+class PublishFileMessageTest : BaseTest() {
+
+    val channel = "coolChannel"
+    val fileId = "fileId"
+    val filename = "filename"
+    val message = "hi"
+
+    @BeforeEach
+    fun before() {
+        stubFor(
+            get(filesUrlPathMatcher()).willReturn(
+                aResponse().withBody("""[1,"Sent","15883272000000000"]""")
+            )
+        )
+    }
+
+    @Test
+    fun testSpaceIdQueryParamIsPassedInPublish() {
+        val spaceIdValue = "thisIsSpaceId"
+
+        pubnub.publishFileMessage(
```

Comment:
I like this test. So wire mock can be used to mimic REST endpoint response but also to check query parameter value/presence.

Reply:
Not sure what you mean by "used to" but I also like approach without using WireMock like in FetchMessagesTest in java where we do:
Call<FetchMessagesEnvelope> call = objectUnderTest.doWork(baseParams);

This approach shows more details about FetchMessage class internals.